cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -302,17 +301,19 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) caerrors.AutoscalerErr
 		klog.Errorf("Failed to get node list: %v", typedErr)
 		return typedErr
 	}
-	originalScheduledPods, err := scheduledPodLister.List()
-	if err != nil {
-		klog.Errorf("Failed to list scheduled pods: %v", err)
-		return caerrors.ToAutoscalerError(caerrors.ApiCallError, err)
-	}
 
 	if abortLoop, err := a.processors.ActionableClusterProcessor.ShouldAbort(
 		a.AutoscalingContext, allNodes, readyNodes, currentTime); abortLoop {
 		return err
 	}
 
+	scheduledAndUnschedulablePods, err := scheduledAndUnschedulablePodLister.List()
+	if err != nil {
+		klog.Errorf("Failed to list scheduled and unschedulable pods: %v", err)
+		return caerrors.ToAutoscalerError(caerrors.ApiCallError, err)
+	}
+	originalScheduledPods, unschedulablePods := seperateScheduledAndUnschedulablePods(scheduledAndUnschedulablePods)
```

Comment:
If the object's responsibility is to list scheduled and unschedulable pods, I don't think it should be the caller's responsibility to distinguish between them, the object should handle that. I'd change the `List` method to return scheduled and unschedulable pods in separate values in the first place.

Reply:
Done